src/FlipMove.js

```diff
@@ -233,6 +231,23 @@ class FlipMove extends Component {
         domNode.style.left  = leavingBoundingBox.left - cleanedComputed['margin-left'] + 'px';
         domNode.style.right = leavingBoundingBox.right - cleanedComputed['margin-right'] + 'px';
       });
+
+      if ( this.props.maintainContainerHeight ) {
+        // We need to find the height of the container *without* the padding
+        // element. Since it's possible that the padding element might already
+        // be present, we first set its height to 0. This allows the container to
+        // collapse down to the size of just its content.
+        this.heightPlaceholder.style.height = 0;
+        const contentHeight = this.props.getPosition(this.parentElement).height;
+
+        // Find the distance by which the container would be collapsed by elements
+        // leaving. We compare the temporarily available `collapsedHeight` with
+        // the previously cached container height.
+        const collapseHeight = this.parentBox.height - contentHeight;
+
+        // Update the padding element's height.
+        this.heightPlaceholder.style.height = Math.max(0, collapseHeight) + 'px';
```

Comment:
Cool. Changed it a bit. It's not exactly like your code above, but hopefully it's explicit enough.

btw, a common way of clamping a number is like this:

```js
function clamp(value, lower, upper) {
  return Math.min(Math.max(value, lower), upper)
}
```